### PR TITLE
CI additions

### PR DESCRIPTION
**Pull request by ghenry**
Nice and clean now. See #202.

### CODE REVIEW

**Comment by ghenry** [comment]
This should be good to go now @davehorton

**Comment by davehorton** [comment]
Thanks.  This looks like it will be helpful -- lots of issues to fix!

Before I merge the changes, I would like to understand how I will be able to run these tests locally on my laptop.  Can you give me some quick pointers?

File: .github/workflows/cppcheck.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '0 9 * * *' # Daily at 10:00 UTC
+jobs:
+  clang-analyzer:
```
Comment (davehorton) [inline .github/workflows/cppcheck.yml:11]: Could you change this name to cpp-analyzer or something, so it doesn't collide with the other "clang-analyzer" job

**Review by davehorton — CHANGES_REQUESTED** [review]

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        env:
+          BUILD_WRAPPER_DOWNLOAD_URL: ${{ env.SONAR_SERVER_URL }}/static/cpp/build-wrapper-linux-x86.zip
+        run: |
+          curl -sSLo $HOME/.sonar/build-wrapper-linux-x86.zip ${{ env.BUILD_WRAPPER_DOWNLOAD_URL }}
+          unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
+          echo "$HOME/.sonar/build-wrapper-linux-x86" >> $GITHUB_PATH
+      - name: build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libtool libtool-bin libcurl4-openssl-dev libpcap-dev
```
Comment (davehorton) [inline .github/workflows/sonarcloud.yml:52]: cmake needs to be added as a dependency

**Review by davehorton — CHANGES_REQUESTED** [review]

File: .github/workflows/sonarcloud.yml (hunk 2 above)
Comment (davehorton) [reply .github/workflows/sonarcloud.yml:52]: scratch that, this is something I need to add in ci.yml.

**Review by davehorton — APPROVED** [review]

**Pull request merged**